packages/vite/src/node/server/middlewares/indexHtml.ts

```diff
@@ -94,9 +94,39 @@ const devHtmlHook: IndexHtmlTransformHook = async (
   const base = config.base || '/'
 
   const s = new MagicString(html)
-  let scriptModuleIndex = -1
+  let inlineModuleIndex = -1
   const filePath = cleanUrl(htmlPath)
 
+  const addInlineModule = (node: ElementNode, ext: 'js' | 'css') => {
+    inlineModuleIndex++
+
+    const url = filePath.replace(normalizePath(config.root), '')
+
+    const contents = node.children
+      .map((child: any) => child.content || '')
+      .join('')
+
+    // add HTML Proxy to Map
+    addToHTMLProxyCache(config, url, inlineModuleIndex, contents)
+
+    // inline js module. convert to src="proxy"
+    const modulePath = `${
+      config.base + htmlPath.slice(1)
+    }?html-proxy&index=${inlineModuleIndex}.${ext}`
+
+    // invalidate the module so the newly cached contents will be served
+    const module = server?.moduleGraph.getModuleById(modulePath)
+    if (module) {
+      server?.moduleGraph.invalidateModule(module)
+    }
+
+    s.overwrite(
+      node.loc.start.offset,
+      node.loc.end.offset,
+      `<script type="module" src="${modulePath}"></script>`
```

Comment:
I think if the type is css, this shouldn't be a `script type="module"`, I think that's causing https://github.com/vitejs/vite/issues/7124.

Reply:
but no module css, it generate code can't run

Reply:
```js
import {createHotContext as __vite__createHotContext} from "/@vite/client";
import.meta.hot = __vite__createHotContext("/dep.css");
import {updateStyle as __vite__updateStyle, removeStyle as __vite__removeStyle} from "/@vite/client"
```